### ISSUE TITLE
Improve scripts by moving from shell module to call curl to the URI module

### ISSUE THREAD

**Issue by pejvan** [jlund/streisand#253]
Hi Joshua,

I think it would be a great step forward to replace shell commands calling cURL by calls to the [URI module](http://docs.ansible.com/ansible/uri_module.html).

Apologies I haven't found any time to dig much deeper into the source and even less to provide an initial fix for this issue as a pull request — hope to be able contribute that some time in the future.

**Comment by cpu** [comment]
I agree this is a worthwhile endeavour. There are some places ([one example](https://github.com/jlund/streisand/blob/d204245ad975eaa8b389c122f130b94ea6c30a4d/playbooks/provider-detect.yml#L38:L46)) where it isn't possible, but overall this should be the method we strive to standardize on.

Pull-requests welcome!
